blockchain-agent/src/main/java/com/sphereon/alfresco/blockchain/agent/config/AlfrescoApisConfig.java

```diff
@@ -94,20 +100,28 @@ SearchApi alfrescoSearchApi() {
     @Bean
     @Scope(ConfigurableBeanFactory.SCOPE_PROTOTYPE)
     ApiClient cmisApiClient() {
-        ApiClient cmisApiClient = new ApiClient();
+        final var cmisApiClient = new ApiClient();
         configureClient(cmisApiClient, CMIS_API_PATH);
         return cmisApiClient;
     }
 
-    private void configureClient(ApiClient apiClient, String apiPath) {
-        HttpBasicAuth authentication = (HttpBasicAuth) apiClient.getAuthentication("basicAuth");
-        authentication.setUsername(userName);
-        authentication.setPassword(password);
+    private void configureClient(final ApiClient apiClient, final String apiPath) {
+        configureClientAuthentication(apiClient);
         apiClient.setBasePath("https://" + alfrescoDnsName + apiPath);
+        configureClientTimeouts(apiClient);
```

Comment:
Suggest to also introduce a configureEndpoint method or something. Don't like the static https and symmetry in passing the apiClient around as well as doing something directly on it here ;)

Reply:
Factored out the literal.

Reply:
Added `configureEndpoint`. Additionally, I must have been sleeping regarding the construction of the clients. In my head they were all different, but I see that they are all `ApiClient`s. So I removed the output parameter on `configureClient` and renamed it to `createClient`.